src/testing/deploy/deploy_test.go

```diff
@@ -102,36 +105,62 @@ func TestEnterpriseServerMember(t *testing.T) {
 }
 
 func mockIDPLogin(t testing.TB, c *client.APIClient) {
-	// login using mock IDP admin
-	hc := &http.Client{}
-	c.SetAuthToken("")
-	loginInfo, err := c.GetOIDCLogin(c.Ctx(), &auth.GetOIDCLoginRequest{})
-	require.NoError(t, err)
-	state := loginInfo.State
+	require.NoErrorWithinTRetryConstant(t, 60*time.Second, func() error {
+		// login using mock IDP admin
+		hc := &http.Client{}
+		c.SetAuthToken("")
+		loginInfo, err := c.GetOIDCLogin(c.Ctx(), &auth.GetOIDCLoginRequest{})
+		if err != nil {
+			return errors.EnsureStack(err)
+		}
+		state := loginInfo.State
 
-	// Get the initial URL from the grpc, which should point to the dex login page
-	getResp, err := hc.Get(loginInfo.LoginURL)
-	require.NoError(t, err)
-	require.Equal(t, 200, getResp.StatusCode, "Mock login could not retrieve the login page.")
+		// Get the initial URL from the grpc, which should point to the dex login page
+		getResp, err := hc.Get(loginInfo.LoginURL)
+		if err != nil {
+			return errors.EnsureStack(err)
+		}
+		if 200 != getResp.StatusCode {
+			return errors.EnsureStack(errors.Errorf("Could not retrieve the mock login page. Expected: 200, Recieved: %d", getResp.StatusCode))
+		}
 
-	vals := make(url.Values)
-	vals.Add("login", "admin")
-	vals.Add("password", "password")
+		vals := make(url.Values)
+		vals.Add("login", "admin")
+		vals.Add("password", "password")
+		postResp, err := hc.PostForm(getResp.Request.URL.String(), vals)
+		if err != nil {
+			return errors.EnsureStack(err)
+		}
+		if 200 != postResp.StatusCode {
+			return errors.EnsureStack(errors.Errorf("POST to perform mock login failed. Expected: 200, Recieved: %d", postResp.StatusCode))
+		}
+		postBody, err := io.ReadAll(postResp.Body)
+		if err != nil {
+			return errors.EnsureStack(errors.Errorf("Could not read login form POST response: %s", err.Error()))
+		}
+		// There is a login failure case in which the response returned is a redirect back to the login page that returns 200, but does not log in
+		expectedResponse := "^(You are now logged in).*$"
+		matches, err := regexp.Match(expectedResponse, postBody)
+		require.NoError(t, err, "Regex matching failed") // a regex compile error should be deterministic, so fail immediately
+		if !matches {
```

Comment:
After you switch to HasPrefix, do this all in one line (`if !strings.HasPrefix(...) { error }`)